plugin-fixes.php

```diff
@@ -18,6 +18,29 @@
 	define( 'WPCF7_UPLOADS_TMP_DIR', get_temp_dir() . 'cf7/' );
 }
 
+/**
+ * Recent versions of CF7 attempt to clean-up uploaded attachments
+ * somewhere within `wp-content`, ignoring the value of WPCF7_UPLOADS_TMP_DIR if
+ * it is not within `wp-content`. This does not work because we disable `open_dir()`,
+ * flooding the PHP logs with `opendir() Failed to open directory` warnings.
+ *
+ * @return void
+ */
+function vip_disable_wpcf7_cleanup_upload_files() {
+	// Return early if the relevant functions do not exist
+	if ( ! function_exists( 'wpcf7_cleanup_upload_files' ) || ! function_exists( 'wpcf7_upload_tmp_dir' ) ) {
+		return;
+	}
+
+	// Check if the action is queued and if the tmp directories match
+	// (they might match if the plugin is an old version, or has been patched)
+	$priority = has_action( 'shutdown', 'wpcf7_cleanup_upload_files' );
+	if ( false !== $priority && WPCF7_UPLOADS_TMP_DIR !== wpcf7_upload_tmp_dir() ) {
```

Comment:
For the sake of extra safety, let's make sure the `WPCF7_UPLOADS_TMP_DIR` constant is defined to avoid potential future PHP errors.

Reply:
+1

Reply:
I will add this, though it is defined a few lines above :)

https://github.com/Automattic/vip-go-mu-plugins/blob/90bd992af476e75c21cc075c012c031f79aeafd7/plugin-fixes.php#L17-L19